### PR TITLE
Add a guide for publishing packages

### PR DESCRIPTION
**Pull request by zanieb**
Closes https://github.com/astral-sh/uv/issues/5793

### CODE REVIEW

**Comment by zanieb** [comment]
Not a ton to say here, can we include more content?

File: docs/guides/publish.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Publishing a package
+
+uv does not yet have dedicated commands for building and publishing a package. Instead, you can use
+the PyPA tools [`build`](https://github.com/pypa/build) and
+[`twine`](https://github.com/pypa/twine).
+
+## Building your package
+
+Build your package with official `build` frontend:
```
Comment (charliermarsh) [inline docs/guides/publish.md:9]: with "the" official?

File: docs/guides/publish.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Publishing a package
+
+uv does not yet have dedicated commands for building and publishing a package. Instead, you can use
+the PyPA tools [`build`](https://github.com/pypa/build) and
+[`twine`](https://github.com/pypa/twine).
```
Comment (charliermarsh) [inline docs/guides/publish.md:5]: "both of which can be accessed via `uvx`."?

**Review by charliermarsh — APPROVED** [review]

File: docs/guides/publish.md (hunk 1 above)
Comment (zanieb) [reply docs/guides/publish.md:9]:
```suggestion
Build your package with the official `build` frontend:
```

File: docs/guides/publish.md (hunk 2 above)
Comment (zanieb) [reply docs/guides/publish.md:5]:
```suggestion
[`twine`](https://github.com/pypa/twine), both of which can be invoked via `uvx`.
```

**Comment by stanmart** [comment]
`build` can use `uv` for creating the build environment. The docs for building the package could make use of this. It's much faster, and imo also conceptually nicer than relying on pip for this step only. The downside is that the build command becomes a bit more complex:

```shell
uvx --from build pyproject-build --installer uv
```

**Comment by zanieb** [comment]
Oh cool, thank you!

**Comment by zanieb** [comment]
https://github.com/astral-sh/uv/pull/5854

**Pull request merged**